### ISSUE TITLE
How to backup and restore freescout

### ISSUE THREAD

**Issue by sainiaj** [freescout-helpdesk/freescout#2568]
Hello i am changing hosting VPS provider and was wondering what are the recommended/official method for backing up everything
purchased modules, data settings etc

Thank you

**Comment by pf248** [comment]
As FS is very lightweigth you can just make a copy of the FS file directory (e.g. /var/www/html/ or whereever your installation is (in this folder there are all the config files and also the module files, etc.). In addition, you can make a database export (e.g. via mysql dump). Then you can zip/tar the two parts of FS and transfer them to a location of your choice. We automated this via a crontab on our ubuntu server to make a backup everyday.

Restore can be done the same way, just move/copy the backup files to the location of your "broken" installation and override all the files. Then do a sql dump in the other direction and FS is up. (When you restore on another server you also have to make sure the requirements such as php are matching and the webserver is configured correctly (here you could also make a copy of the apache/nginx file and transfer it))

Hope this helps you little bit.

**Comment by freescout-helpdesk** [comment]
@pf248 thanks for helping out.

**Comment by freescout-helpdesk** [comment]
https://github.com/freescout-helpdesk/freescout/wiki/FAQ#how-to-backup-and-restore-the-freescout

**Comment by sainiaj** [comment]
@freescout-helpdesk Yes as mentioned by @fulldecent in the future please implement backup and restore feature it is a must, i cannot even backup the sql as i don't have the password saved,

until then please make official documentation how to use msql dump cmd and save 
new install > restore > and apply ?

**Comment by aj-saini** [comment]
@pf248 pf248
Would it be possible to share the script
you use to backup your free scout daily to ubuntu server - i would like to setup something like that as well

Thank you

**Comment by pf248** [comment]
@aj-saini Yes, of course. I took the backup & restore scripts from here: https://codeberg.org/DecaTec/Wordpress-Backup-Restore

It is for wordpress, but you can just change the parameters in the shell scripts to your data (like db-user, password, path to the installation, backup directory, etc.). It is all described in the files, quite easy. The backup script does a copy of the installation directory and also a sql datebase dump. You can control how many backups you want to keep and the backup script automatically deletes all backups older then a specific amount (e.g. keep the last 10 backups).

Then you just set up a crontab on your ubuntu system to execute the backup.sh (e.g. every night at 1:00). Command could be sth like:
 * 1  *  *   *   cd /path_to_file && ./backup.sh

The backup.sh must be executable for this, you can do this via chmod +x.

Hope this helps you a bit.

**Comment by semaf** [comment]
Why is the documentation for backup so poor? There is nothing for docker to backup.

**Comment by NBSUKLTD** [comment]
Agreed a freescout backup module with scheduling and remote SFTP, FTP, S3 storage options would be great.

**Comment by jacobgraf** [comment]
> Agreed a freescout backup module with scheduling and remote SFTP, FTP, S3 storage options would be great.

Yes, please!
